plutus-ledger-api/src/PlutusLedgerApi/Common/Versions.hs

```diff
@@ -153,44 +104,181 @@ ledgerLanguageIntroducedIn = \case
+{- It's tempting to try something like `fmap toEnum [0..50]` here, but that's
+   dangerous because the order of the constructors in DefaultFun doesn't
+   precisely match the order that the builtins were introduced in.  A safer
+   alternative would be to use the flat tags, but they're not directly
+   accessible at the moment.
+-}
+-- DO NOT CHANGE THIS.
+batch1 :: [DefaultFun]
+batch1 =
+  [ AddInteger, SubtractInteger, MultiplyInteger, DivideInteger, QuotientInteger
+  , RemainderInteger, ModInteger, EqualsInteger, LessThanInteger, LessThanEqualsInteger
+  , AppendByteString, ConsByteString, SliceByteString, LengthOfByteString
+  , IndexByteString, EqualsByteString, LessThanByteString, LessThanEqualsByteString
+  , Sha2_256, Sha3_256, Blake2b_256, VerifyEd25519Signature, AppendString, EqualsString
+  , EncodeUtf8, DecodeUtf8, IfThenElse, ChooseUnit, Trace, FstPair, SndPair, ChooseList
+  , MkCons, HeadList, TailList, NullList, ChooseData, ConstrData, MapData, ListData
+  , IData, BData, UnConstrData, UnMapData, UnListData, UnIData, UnBData, EqualsData
+  , MkPairData, MkNilData, MkNilPairData
+  ]
+
+-- DO NOT CHANGE THIS.
+batch2 :: [DefaultFun]
+batch2 =
+  [ SerialiseData ]
+
+-- DO NOT CHANGE THIS.
+batch3 :: [DefaultFun]
+batch3 =
+  [ VerifyEcdsaSecp256k1Signature, VerifySchnorrSecp256k1Signature ]
+
+-- `cekCase` and `cekConstr` costs come between Batch 3 and Batch 4 in the
+-- PlutusV3 cost model parameters, although that's irrelevant here.
+
+-- batch4, excluding IntegerToByteString and ByteStringToInteger.
+-- DO NOT CHANGE THIS.
+batch4a :: [DefaultFun]
+batch4a =
+  [ Bls12_381_G1_add, Bls12_381_G1_neg, Bls12_381_G1_scalarMul
+  , Bls12_381_G1_equal, Bls12_381_G1_hashToGroup
+  , Bls12_381_G1_compress, Bls12_381_G1_uncompress
+  , Bls12_381_G2_add, Bls12_381_G2_neg, Bls12_381_G2_scalarMul
+  , Bls12_381_G2_equal, Bls12_381_G2_hashToGroup
+  , Bls12_381_G2_compress, Bls12_381_G2_uncompress
+  , Bls12_381_millerLoop, Bls12_381_mulMlResult, Bls12_381_finalVerify
+  , Keccak_256, Blake2b_224
+  ]
+
+{- batch4b: IntegerToByteString and ByteStringToInteger.  These were enabled in
+ PlutusV3 at PV9, along with batch4a, They were enabled in PlutusV2 at PV10 in
+ #6056 and #6065.  They are available on the chain, but they're prohibitively
+ expensive because the proposal to update the relevant protocol parameters has
+ not (yet) been enacted.  This has left a "gap" in the cost model paramters: for
+ PlutusV3, the parameters for Batch 3 are followed those for 4a, then 4b, but
+ for PlutusV2 those for Batch3 are followed by those for Batch 4a, and those for
+ 4b aren't in use yet.  Since you can't actually use the 4b builtins in PlutusV2
+ at the moment, it's tempting to insert the 4a parameter before the 4b
+ parameters and enable them all at PV11 and with a suitable parameter update.
+ However, if we do do this there's a theoretical risk of turning a phase 2
+ failure into a phase 1 failure: would that be problematic?
```

Comment:
Good point - different node versions disagreeing on whether a failure is phase 1 or phase 2 can indeed be _very_ problematic.